src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogHttpClient.java

```diff
@@ -124,6 +124,70 @@ private DatadogHttpClient(String url, String logIntakeUrl, Secret apiKey) {
         this.logIntakeUrl = logIntakeUrl;
     }
 
+    public void validateConfiguration() throws IllegalArgumentException {
+        if (url == null || url.isEmpty()) {
+            throw new IllegalArgumentException("Datadog Target URL is not set properly");
+        }
+        if (apiKey == null || Secret.toString(apiKey).isEmpty()){
+            throw new IllegalArgumentException("Datadog API Key is not set properly");
+        }
+        if (DatadogHttpClient.isCollectBuildLogEnabled() && (logIntakeUrl == null || logIntakeUrl.isEmpty())){
+            throw new IllegalArgumentException("Datadog Log Intake URL is not set properly");
+        }
+        try {
+            boolean intakeConnection = validateDefaultIntakeConnection(url, apiKey);
+            if (!intakeConnection) {
+                instance.setDefaultIntakeConnectionBroken(true);
+
+                throw new IllegalArgumentException("Connection broken, please double check both your API URL and Key");
+            }
+        } catch (IOException e) {
+            instance.setDefaultIntakeConnectionBroken(true);
+            throw new IllegalArgumentException("Connection broken, please double check both your API URL and Key");
+        }
+        if (DatadogHttpClient.isCollectBuildLogEnabled()) {
+            try {
+                boolean logConnection = validateLogIntakeConnection(url, apiKey);
+                if (!logConnection) {
+                    instance.setLogIntakeConnectionBroken(true);
+                    throw new IllegalArgumentException("Connection broken, please double check both your Log Intake URL and Key");
+                }
+            } catch (IOException e) {
+                instance.setLogIntakeConnectionBroken(true);
+                throw new IllegalArgumentException("Connection broken, please double check both your Log Intake URL and Key");
+            }
+        }
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == this) {
+            return true;
+        }
+        if (!(object instanceof DatadogClient)) {
+            return false;
+        }
+
+        DatadogHttpClient newInstance = (DatadogHttpClient) object;
+        DatadogHttpClient httpInstance = (DatadogHttpClient) instance;
+
+        if ((StringUtils.equals(httpInstance.getLogIntakeUrl(), newInstance.getLogIntakeUrl()))
+        && (StringUtils.equals(httpInstance.getUrl(), newInstance.getUrl())
+        && ((newInstance.getApiKey() == null && httpInstance.getApiKey() == null) || newInstance.getApiKey().equals(httpInstance.getApiKey())))){
```

Comment:
```suggestion
        DatadogHttpClient newInstance = (DatadogHttpClient) object;

        if ((StringUtils.equals(getLogIntakeUrl(), newInstance.getLogIntakeUrl()))
                && (StringUtils.equals(getUrl(), newInstance.getUrl())
                && ((newInstance.getApiKey() == null && getApiKey() == null) || newInstance.getApiKey().equals(getApiKey())))) {
```

Not sure if this is possible, from my previous comment regarding the static instance variable - but if we can avoid doing the `httpInstance` cast, then the comparison code becomes much simpler.

Reply:
It actually is possible without casting, now! it was unnecessary before since all of the methods called were in `DatadogClient` so I removed it.